File: Post/DrQuine/Lua/Colleen.lua
-- This program will print its own source when run.
Colleen = "-- This program will print its own source when run.%1Colleen = %2%3%2%1%1function print(s)%1	io.write(s)%1end%1%1function main()%1	-- This program will print its own source when run.%1	print(Colleen:gsub(%2%%%%.%2, {[%2%%%%%2] = '%%', [%2%%1%2] = string.char(10), [%2%%2%2] = '%2', [%2%%3%2] = Colleen}))%1end%1%1main()%1"

function print(s)
	io.write(s)
end

function main()
	-- This program will print its own source when run.
	print(Colleen:gsub("%%.", {["%%"] = '%', ["%1"] = string.char(10), ["%2"] = '"', ["%3"] = Colleen}))
end

main()

File: Post/DrQuine/C/Colleen.c
#include <stdio.h>
#define COLLEEN "#include <stdio.h>%2$c#define COLLEEN %3$c%1$s%3$c%2$c%2$cvoid out()%2$c{%2$c	printf(COLLEEN, COLLEEN, 10, 34);%2$c}%2$c%2$cint main()%2$c{%2$c	out();%2$c/*%2$c	This program will print its own source when run.%2$c*/%2$c}%2$c/*%2$c	This program will print its own source when run.%2$c*/%2$c"

void out()
{
	printf(COLLEEN, COLLEEN, 10, 34);
}

int main()
{
	out();
/*
	This program will print its own source when run.
*/
}
/*
	This program will print its own source when run.
*/
